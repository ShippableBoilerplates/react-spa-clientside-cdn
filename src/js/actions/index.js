import ACTIONS from '../constants'

export default {
    updateClickCount: (num) => {
        return {
            type: ACTIONS.UPDATE_CLICK_COUNT,
            payload: num
        }
    },
    getChuckNorrisJoke: (joke) => {
        return {
            type: ACTIONS.GET_CHUCK_NORRIS_JOKE,
            payload: joke
        }
    }
}
