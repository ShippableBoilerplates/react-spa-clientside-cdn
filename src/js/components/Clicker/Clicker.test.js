import React from 'React'
import {shallow} from 'enzyme'
import {Clicker} from './Clicker'
import {default as action} from '../../actions/'

test('should render Clicker component correctly', () => {
    const wrapper = shallow(
        <Clicker
            updateClickCount={action.updateClickCount}
            counter={0}
            joke={''}
        />)
    expect(wrapper).toMatchSnapshot()
})
