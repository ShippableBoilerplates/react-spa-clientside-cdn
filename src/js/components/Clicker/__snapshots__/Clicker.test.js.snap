// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Clicker component correctly 1`] = `
<div
  data-css-1duqoor=""
>
  <div
    data-css-1gqc584=""
  >
    <img
      data-css-pi5t0l=""
      onClick={[Function]}
      src="react.svg"
    />
    <p
      data-css-1khb4zr=""
    >
      You split the atom 
      0
       times!
    </p>
    <p
      data-css-x7041l=""
    />
  </div>
</div>
`;
